lib/puppet/provider/port_channel/cisco_nexus.rb

```diff
@@ -3,39 +3,60 @@
 
 # Implementation for the port_channel type using the Resource API.
 class Puppet::Provider::PortChannel::CiscoNexus < Puppet::ResourceApi::SimpleProvider
-  def get(_context)
+  def canonicalize(_context, resources)
+    resources
+  end
+
+  def get(_context, channels=nil)
     interfaces = []
```

Comment:
require 'cisco_node_utils'
here?

Reply:
Good spot @willmeek, and remove it from the top of file on line 2 and hopefully there won't be many other places.

Reasoning is they would be forced to include the gem on their puppetserver which we do not want. 

Similar too the reasoning for: https://tickets.puppetlabs.com/browse/FM-7496 where more information is provided

Reply:
Updated the providers. Removed any reference of `require 'cisco_node_utils'` from the top of the provider and moved it to `get`. Also added to `canonicalize` if required.

Reply:
```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Could not autoload puppet/type/port_channel: Could not autoload puppet/provider/port_channel/cisco_nexus: no such file to load -- cisco_node_utils (file: /etc/puppetlabs/code/environments/production/manifests/site.pp, line: 50, column: 3) on node n9kv1
```

This is the type of error which would be expected when the `cisco_node_utils` is not included on the puppetserver, by removing it from the top of file and only having it in an initialise or the `get` will avoid this encounter.

Reply:
> Updated the providers. Removed any reference of `require 'cisco_node_utils'` from the top of the provider and moved it to `get`. Also added to `canonicalize` if required.

`canonicalize` wouldn't require it but makes no harm in including it I suppose.